internal/routing/local.go

```diff
@@ -85,3 +90,14 @@ func (r *Routing) LocalNetworks() (localNetworks []LocalNetwork, err error) {
 
 	return localNetworks, nil
 }
+
+func (r *Routing) AddLocalRules(subnets []LocalNetwork) (err error) {
+	for _, net := range subnets {
+		// Main table was setup correctly by Docker, just need to add rules to use it
+		err = r.addIPRule(nil, net.IPNet, mainTable, localPriority)
+		if err != nil {
+			return fmt.Errorf("cannot add rule: for subnet %s: %w", net.IPNet, err)
```

Comment:
nit on the error (note the existing code may or may not follow this, but new code should):

- do not prefix context with `cannot ` (or similar) otherwise it repeats on errors wrapped multiple times, i.e. `cannot do X: cannot do Y: actual error`. Instead use the form `doing X: doing Y: actual error`
- the subnet `net.IPNet` is already passed down `addIPRule` so it should be the reponsibility of `addIPRule` to add it to the error context, to avoid duplicating the context. In that case it's part of the context as the rule's destination (`fmt.Errorf("cannot add rule %s: %w", rule, err)` + `rule.Dst = dst`)

All this leading to:

```suggestion
			return fmt.Errorf("adding rule: %w", net.IPNet, err)
```

Reply:
Thanks for the feedback. Updated as requested.